### DISCUSSION TITLE
Inconsistent behavior of brew livecheck and brew outdated/upgrade

### DISCUSSION THREAD

**Discussion by ChristianKraemer** [Homebrew/discussions#4228] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.1-34-g8ff2e29
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8ff2e29aadc1e3d08b14acf0c45664e593eecc86
Last commit: 4 hours ago
Core tap JSON: 17 Feb 10:27 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.90SySJNQtD/org.xquartz:0
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_GOOGLE_ANALYTICS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.2 => /usr/local/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

What happened:

With the release of 4.0.0 yesterday I followed the advises and removed this NO_INSTALL_FROM_API thing and untapped core and cask.

An update to microsoft-edge appeared and did a bump-cask-pr. To be able to do so I had to tap homebrew/cask again. (By the way, the underlying git operation seemed much slower - showing speed in KiB instead of MiB before - coincidence or result of using personal API token?) 

Over night the PR was merged and an "brew upgrade" this morning updated microsoft-edge.

Situation now:

"brew outdated --greedy" and "brew upgrade --greedy" do not show any output.

"brew lc --newer-only --installed" shows the line:

"microsoft-edge: 110.0.1587.41 ==> 110.0.1587.49"

Version 110.0.1587.49 is that update I installed in the morning.

How can I convince "brew lc" that microsoft-edge is up to date?

**Comment by SMillerDev** [comment]
What does `brew info microsoft-edge` say?

**Comment by ChristianKraemer** [comment]
Now that I collected all facts to write it down I tried something. A cd $(brew --repo homebrew/cask) and "git pull" there lead to the expected behavior. "brew lc" recognizes the up to date microsoft-edge.

**Reply by cho-m** [reply]
`brew livecheck` relies on tap to function and uses information from local tapped git repo.

Common `brew` commands now default to API and ignore the git repo. So, without `HOMEBREW_NO_INSTALL_FROM_API`, the local repos are ignored. This avoids the overhead `git pull` but can cause git repos to be outdated.

If you are planning to commonly run `brew livecheck` and submit bump PRs, then it may be simpler to not use API.

---

EDIT: Maybe another option would be for `brew` to force update the git repos with certain commands, but would need to figure out if feasible and if there are any scenarios this would break.

**Comment by ChristianKraemer** [comment]
Result of "brew info microsoft-edge" is:

==> Downloading https://formulae.brew.sh/api/formula.json
######################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.json
######################################################################## 100.0%
==> microsoft-edge: 110.0.1587.49 (auto_updates)
https://www.microsoft.com/edge
/usr/local/Caskroom/microsoft-edge/110.0.1587.49 (205MB)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/microsoft-edge.rb
==> Name
Microsoft Edge
==> Description
Web browser
==> Artifacts
MicrosoftEdge-110.0.1587.49.pkg (Pkg)
==> Analytics
install: 4,730 (30 days), 10,374 (90 days), 41,414 (365 days)
